### DISCUSSION TITLE
OpenEBS Helm chart broken: installs resources into multiple namespaces (default + custom)

### DISCUSSION THREAD

**Discussion by svengreb** [openebs/openebs#3760] [Q&A]
I tried to install [the OpenEBS Helm chart (version `4.1.0`)][1] into a custom `openebs` namespace, but it seems like some templates do not set the `.metadata.namespace`, neither through `.Release.Namespace` nor through a custom value. This results in some resources being created in the `default` namespace, e.g. `serviceaccount/openebs-localpv-provisioner` while `serviceaccount/openebs-pre-upgrade-hook` is created in the desired (`openebs`) namespace.

I'm very experienced with Helm and tested a lot of settings, but I also wonder why this is not a problem anyone is facing too (checked GitHub issues and other discussions). Simply following [the official Helm installation documentation][6] will result in this broken setup so it should be noticed somehow.

Here's the command I used to check the resulting manifests, using Helm's `template` command:

```shell
helm template openebs openebs/openebs --version 4.1.0 --namespace openebs --create-namespace
```

The `openebs` Helm chart version 4.1.0 uses [version `4.0.0`][3] of [the `dynamic-localpv-provisioner` Helm chart][4] where e.g. [the deployment is missing the `namespace` key][5] so it will be deployed into the `default` namespace.

My current workaround is to use [the _Kustomize_ Helm feature][2] that sets the namespace for all resources that are namespaced.

```yaml
# kustomization.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

namespace: openebs

helmCharts:
  - name: openebs
    namespace: openebs
    version: 4.1.0
    includeCRDs: true
    releaseName: openebs
    repo: https://openebs.github.io/openebs
    valuesInline:
      # …
```

So to summarize my questions: Is this a know problem or did I miss something, e.g. a value key where the target namespace for each subchart can be customized?

[1]: https://github.com/openebs/openebs/tree/v4.1.0/charts
[2]: https://github.com/kubernetes-sigs/kustomize/blob/v3.3.1/examples/chart.md
[3]: https://github.com/openebs/openebs/blob/v4.1.0/charts/Chart.yaml#L20-L22
[4]: https://github.com/openebs/dynamic-localpv-provisioner/tree/v4.0.0/deploy/helm/charts
[5]: https://github.com/openebs/dynamic-localpv-provisioner/blob/v4.0.0/deploy/helm/charts/templates/deployment.yaml#L4-L10
[6]: https://openebs.io/docs/quickstart-guide/installation#installation-via-helm

**Comment by niladrih** [comment]
All of the resources from all of the helm charts in the openebs scope go to the helm release namespace. The namespace cannot be set explicitly for dependencies.
